### DISCUSSION TITLE
[BUG]Plot always crash at 68% (bucket 57)

### DISCUSSION THREAD

**Discussion by kikocaramelo** [Chia-Network/chia-blockchain#7406] [Support]
Plot always crash at 68% (bucket 57)
I'm in Chia 1.2.0, plotting for a pool. I'm using a Mac mini Intel withMacOs 10.15.7
I tried several plots all blocked.
I'm also using a PC with the same wallet and it works, everything OK on the PC.

**Comment by loppefaaret** [comment]
around 68% sounds like you are into phase3, where memory usage is topping - are you potentially running out of memory ?
could you provide the last few lines of the failed plotting log ? that could provide valuable information
turning this into a support discussion for now

**Comment by kikocaramelo** [comment]
I make new polo in 1.2.2 and it work 100%. I think the bug is dead.

**Comment by kikocaramelo** [comment]
I partied too early. It's stopping again. Now at 78% (bucket 79).

Bucket 77 uniform sort. Ram: 4.850GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 77 uniform sort. Ram: 4.850GiB, u_sort min: 0.500GiB, qs min: 0.250GiB.
	Bucket 78 uniform sort. Ram: 4.850GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 78 uniform sort. Ram: 4.850GiB, u_sort min: 0.500GiB, qs min: 0.250GiB.
	Bucket 79 uniform sort. Ram: 4.850GiB, u_sort min: 1.250GiB, qs min: 0.314GiB.
	Bucket 79 uniform sort. Ram: 4.850GiB, u_sort min: 0.500GiB, qs min: 0.250GiB.

**Comment by SlugFiller** [comment]
Try reducing the RAM settings. The RAM setting is used to determine how large of a chunk would be sorted via uniform sort (done in-memory). Any chunk larger than that is sorted using quicksort (on disk). The on-disk sort is twice as slow as in-memory, but there's no beating having to restart the plotting because it crashed with a memory error.

You can also consider plotting with MadMax instead.

**Comment by kikocaramelo** [comment]
I reduce the RAM memory and make in other Disk, now stop at 68%
I have 16RAM. At first I put 10RAM, this time I put 6RAM.
It don't alo put more buckets than 128, I tried 256 but it give me error at start.

I only use MadMAx on the PC, I don't found a easy way to use MadMAx in the MAC. Do you now how?

Bucket 73 uniform sort. Ram: 7.753GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 74 uniform sort. Ram: 7.753GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 75 uniform sort. Ram: 7.753GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 76 uniform sort. Ram: 7.753GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 77 uniform sort. Ram: 7.753GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.
	Bucket 78 uniform sort. Ram: 7.753GiB, u_sort min: 1.250GiB, qs min: 0.313GiB.

**Reply by SlugFiller** [reply]
If you can plot on PC, you can just copy a plot to the Mac after it finishes plotting, assuming you're using an internal drive that can't be connected to the PC directly. The plots don't have a different format between Mac and PC, so you should plot on whichever machine gives you the quickest and most stable results.

**Comment by rmica** [comment]
Once I've had the same failure and it ended up being a hardware malfunction. First I blamed the nvme, changed it, the freeze recurred, always at some point in phase 3. It was the motherboard. Something in the PCIe bus was broken. I'd try the standard plotter with default mem/thread settings as a confirmation test.
My (ubuntu) main symptoms were: process running out of any syscall (strace), positive nvme error-log (but it wasn't a nvme failure), accumulating writes in /proc/diskstats.
One very important information is: did your box succesfully plot before and stopped working or did it fail at first ever run? In the first case I'd go for hw error.
Just one more tip: check system and nvme temperature.
